### DISCUSSION TITLE
How does tauri compare to rust-sciter?

### DISCUSSION THREAD

**Discussion by dclong** [tauri-apps/tauri#2007] [Q&A]
It seems to me that rust-sciter is very similar to tauri. Both uses Rust + Frontend technologies. I wonder what are the advantages and disadvantages of tauri compared to rust-sciter?

**Comment by nothingismagick** [comment]
Well, the big one that sticks out to me is the fact that the licensing is not free if you want to build for all three desktop platforms. Furthermore, if you want source you have to pay even more. See https://sciter.com/prices/

 Tauri is and will always remain open source and MIT / Apache-2.0 licensed - even when we support iOS and Android. Our organization exists in the context of the Commons Conservancy, in order to guarantee that none of us go rogue and that bus factor can't stop us. Furthermore, being opensource, means that collaborators can help the community with code revisions instead of merely filing tickets and praying.

Going back to sciter.rs, on consultation of the rust bindings, it seems that menuing and taskbars seem to be overlooked, as does 'always on top'. Perhaps it is the fact that I haven't gone deep enough, but it also appears that the HTML5 / CSS / JS engines are handcrafted, which is a little frightening - considering the amount of things that can go wrong with webengines.

To be clear, sciter.rs is 'merely' bindings to an underlying C architecture, which is very likely to have human error, memory leaks, and undefined behaviour. Tauri is written almost 100% in stable rust, which means that our friend the compiler stomps out those general kinds of problems by its very nature.

That said, sciter appears to have been around for about 15 years, and very obviously has had time to mature. 5MB for a basic app is really quite good, although at tauri we consistently see final binaries that are half that size.


I did not find a recent security audit, which is going to happen for tauri not only this summer, but also on a regular and ongoing basis.

**Reply by dclong** [reply]
Sounds like Tauri is a much better alternative than rust-sciter.
